### PR TITLE
Ft login - setup complete

### PR DESCRIPTION
**Pull request by Brendajoshua**
# Description

Add attribution link to Readme and code

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

## Change status

- [x] Complete, but not tested (may need new tests)

# How Has This Been Tested?

Please describe steps taken to ensure this feature is functional and does not break other features:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation, if relevant
- [x] I have run the app using my feature and ensured that no functionality is broken
- [x] My changes generate no new warnings
- [x] There are no merge conflicts

### CODE REVIEW

File: api/controllers/user.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  },
+  async loginUser (req, res) {
+    const user = await User.findOne({ email: req.body.email }).exec();
+    const login = user.comparePassword(req.body.password);
+    if(!login) {
+      res.status(404).json({
+        error: 'invalid credentials'
+      });
+    }
+    return {...login, token: generateToken(login)};
```
Comment (iamcristos) [inline api/controllers/user.js]: kindly return res.status(200).json({user:login, token: generateToken(login)})

File: api/routes/authroute.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const express = require("express");
+const authUser = require('../helpers/jwt');
+const controller = require('../controllers/user');
+const userValidators = require('../validation/userValidator');
+
+const router = express.Router();
+// eslint-disable-next-line no-unused-vars
+router.post("/signup",[userValidators.validateUserOnSignup], controller.createUser);
+
```
Comment (iamcristos) [inline api/routes/authroute.js]: also hook login route here

**Comment by Brendajoshua** [comment]
I've made the changes to the code, please review for merging.

File: api/controllers/user.js [hunk 3]
```diff
@@ -42,5 +45,25 @@ module.exports = {
     } catch (error) {
       return next(error);
     }
+  },
+  async loginUser (req, res, next) {
+    try{
+    const user = await User.findOne({ email: req.body.email }).exec();
+    if(!user) {
+      return errorHelper(res, 401, `The email address ${req.body.email} is not associated with any account. Double-check your email address and try again.`);
```
Comment (iamcristos) [inline api/controllers/user.js]: can you remove this we don't want to have hackers knowing when they have the right registered email on our platform

Comment (Brendajoshua) [reply api/controllers/user.js]: I have edited that.

**Pull request merged**